### PR TITLE
Build waitlist screens and routes

### PR DESCRIPTION
**Pull request by Excellent-coding**
https://linear.app/find-labs/issue/ENG-17/build-waitlist-screens-and-routes

### CODE REVIEW

File: apps/web-app/src/pages/waitlist.tsx [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 import Button from "../components/Button";
-import SocialLogin from "../components/SocialLogin"
 import { z } from 'zod'
 
 const JoinWaitlist = () => {
   const [email, setEmail] = useState('');
+  const [error, setError] = useState('');
   const [loading, setLoading] = useState(false);
 
   const handleSubmit = () => {
```
Comment (khalil-omer) [inline apps/web-app/src/pages/waitlist.tsx:10]: This function should actually handle submit by calling a tRPC query. https://trpc.io/docs/react-queries

Then we also need to define that query in the Next.js backend via trpc. We can use Prisma to update the waitlist table.

**Review by khalil-omer — CHANGES_REQUESTED** [review]

**Comment by Excellent-coding** [comment]
trpc waitlist table
waitlist create or update by email

File: apps/web-app/prisma/schema.prisma [hunk 2]
```diff
@@ -58,3 +58,8 @@ model VerificationToken {
 
     @@unique([identifier, token])
 }
+
+model Waitlist {
```
Comment (khalil-omer) [inline apps/web-app/prisma/schema.prisma:62]: We already discussed this schema. We don't need an id. We can just use email. And we need to know some other values.

```
model Waitlist {
    // Email address that wants to join Find
    email String @id
    // When user submitted the waitlist form
    createdAt DateTime @default(now())
    modifiedAt DateTime @updatedAt
    // If the waitlist entry has already been converted to a user.
    redeemed Boolean @default(false)
    // Whether to allow a user to join with the redemption secret
    allowRedeem Boolean @default(false)†
}
```

**Review by khalil-omer — CHANGES_REQUESTED** [review]

File: apps/web-app/prisma/schema.prisma (hunk 2 above)
Comment (Excellent-coding) [reply apps/web-app/prisma/schema.prisma:62]: yes, I will update the schema

File: apps/web-app/prisma/schema.prisma [hunk 3]
```diff
@@ -7,7 +7,7 @@ generator client {
 
 datasource db {
     provider = "postgresql" 
-    url      = env("DATABASE_URL")
+    url      = "postgres://postgres:mysecretpassword@localhost:5432/find"
```
Comment (khalil-omer) [inline apps/web-app/prisma/schema.prisma:10]: This needs to stay an env. Revert this change please.

**Review by khalil-omer — CHANGES_REQUESTED** [review]
Env

**Review by khalil-omer — APPROVED** [review]

**Pull request merged**